ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -692,3 +721,132 @@ func validateTaskNetworkConfig(taskNetworkConfig *state.TaskNetworkConfig) error
 
 	return nil
 }
+
+// checkPacketLossFault checks if there's existing network-packet-loss fault running.
+func (h *FaultHandler) checkPacketLossFault(taskMetadata *state.TaskResponse, request types.NetworkPacketLossRequest) (bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	interfaceName := taskMetadata.TaskNetworkConfig.NetworkNamespaces[0].NetworkInterfaces[0].DeviceName
+	lossPercent := request.LossPercent
+	networkMode := taskMetadata.TaskNetworkConfig.NetworkMode
+	containers := taskMetadata.Containers
+	ipSources := request.Sources
+	// If task's network mode is awsvpc, we need to run nsenter to access the task's network namespace.
+	nsenterPrefix := ""
+	if networkMode == "awsvpc" {
+		nsenterPrefix = fmt.Sprintf(nsenterCommandString, containers[0].DockerName)
```

Comment:
IIRC we can get the full path from task metadata.

Reply:
Yes, the full path will be read from the metadata. This is to concatenate the `nsenter --net=` prefix